src/analysis/retail/evoker/augmentation/modules/features/BuffTargetHelper/BuffTargetHelper.tsx

```diff
@@ -216,7 +218,9 @@ class BuffTargetHelper extends Analyzer {
 
     const fetchPromises: Promise<DamageTables>[] = [];
     while (currentTime < this.fightEnd) {
```

Comment:
After looking at this with some rested eyes, I realize that this while loop should never actually be able to broken out of unless you somehow manage to apply an EM on the same tick that the fight ends.
It should be made into some sort of iteration across the timestamp arrays.
eg.
```ts
    for (let i = 0; i < this.ebonApplyTimestamps.length; i += 1) {
       fetchPromises.push(this.getDamage(this.ebonApplyTimestamps[i], this.ebonRemoveTimestamps[i]))
    }
```
The while loop will also end up trying to index undefined fields, so you could do an undefined check and break the loop based on that, but a for loop makes more sense.